engine/src/configuration/configuration.cpp

```diff
@@ -35,35 +38,33 @@
 
 
 std::string GetString(std::string json_string, const std::string key, bool trim = true) {
-    json::jobject json = json::jobject::parse(json_string);
+    // parse the JSON data
+    boost::json::value json_value = boost::json::parse(json_string);
+    boost::json::object root = json_value.get_object();
+
+    // key can be a set of keys delimited by `|` so traverse
+    // the JSON structure through the list of keys
+    // boost::json support using `/` as a delimiter though
     std::vector<std::string> key_sections;
     boost::split(key_sections, key, boost::is_any_of("|"));
-    int key_sections_size = key_sections.size();
-
-    for(std::string& key_section : key_sections) {
-        key_sections_size--;
-
-        if(!json.has_key(key_section)) {
-            return "";
-        }
 
-        if(key_sections_size > 0) {
-            json_string = json.get(key_section);
-            json = json::jobject::parse(json_string);
-        } else {
-            json_string = json.get(key_section);
+    // rebuild the key_sections as a list using `/` per Boost::JSON
+    std::string new_key = boost::algorithm::join(key_sections, "/");
 
-            if(!trim) {
-                return json_string;
-            }
+    std::string value = JsonGetStringWithDefault(root, new_key, "");
```

Comment:
This function does not do what you expect it to do. It does not automatically iterate over several json layers. It probably should. But right now, this code fails to work.

Reply:
That would be appreciated. It'll be a bit until I can build and test again